File: frontend/src/types/permissionAdmin.ts
/**
 * 权限管理相关类型定义
 */

export type PrivilegeType = 
  | 'SELECT'
  | 'INSERT'
  | 'UPDATE'
  | 'DELETE'
  | 'CREATE'
  | 'DROP'
  | 'ALTER'
  | 'INDEX'
  | 'REFERENCES'
  | 'TRIGGER'
  | 'EXECUTE'
  | 'USAGE'

export interface PermissionNode {
  type: 'database' | 'table' | 'column'
  name: string
  path: string
  privileges: PrivilegeType[]
  children?: PermissionNode[]
}

export interface GrantPermissionRequest {
  username: string
  host?: string // MySQL only
  target_type: 'database' | 'table' | 'column'
  target_name: string
  database?: string
  table?: string
  privileges: PrivilegeType[]
}

export interface RevokePermissionRequest {
  username: string
  host?: string // MySQL only
  target_type: 'database' | 'table' | 'column'
  target_name: string
  database?: string
  table?: string
  privileges: PrivilegeType[]
}

File: backend/internal/models/permission_admin.go
package models

// GrantPermissionRequest 授予权限请求
type GrantPermissionRequest struct {
	Username    string   `json:"username" binding:"required"`
	Host        string   `json:"host,omitempty"` // MySQL only
	TargetType  string   `json:"target_type" binding:"required"` // database, table, column
	TargetName  string   `json:"target_name" binding:"required"`
	Database    string   `json:"database,omitempty"` // for table/column
	Table       string   `json:"table,omitempty"`    // for column
	Privileges  []string `json:"privileges" binding:"required"`
}

// RevokePermissionRequest 撤销权限请求
type RevokePermissionRequest struct {
	Username   string   `json:"username" binding:"required"`
	Host       string   `json:"host,omitempty"` // MySQL only
	TargetType string   `json:"target_type" binding:"required"`
	TargetName string   `json:"target_name" binding:"required"`
	Database   string   `json:"database,omitempty"`
	Table      string   `json:"table,omitempty"`
	Privileges []string `json:"privileges" binding:"required"`
}
